### PR TITLE
Modified to use dataset and pipelinedata

### PR DESCRIPTION
**Pull request by GraemeMalcolm**
I modified the pipeline to be more "AMLish":

- Added instructions to Getting Started to register a dataset (or set variables for the pipeline to do it for you)
- Used input in training step to pass dataset object
- Added pipelinedata object to pass trained model as output from train step and input to register step

### CODE REVIEW

File: diabetes_regression/register/register_model.py [hunk 1]
```diff
@@ -109,35 +136,28 @@ def model_already_registered(model_name, exp, run_id):
...
-                             "run_id": run_id, "experiment_name": exp.name}
-            else:
-                print("A model run for experiment", exp.name,
-                      "matching properties run_id =", run_id,
-                      "was not found. Skipping model registration.")
-                sys.exit(0)
+        model_already_registered(model_name, exp, run_id)
+        tagsValue = {
+            "area": "diabetes_regression",
+            "BuildId": build_id,
```
Comment (dtzar) [inline diabetes_regression/register/register_model.py]: This value may not exist.  The logic you removed prevented logging a 'none' build_id from line 101.

File: docs/getting_started.md [hunk 2]
```diff
@@ -60,6 +60,8 @@ The variable group should contain the following required variables:
 | WORKSPACE_NAME           | mlops-AML-WS             |
 | WORKSPACE_SVC_CONNECTION | aml-workspace-connection |
 | ACI_DEPLOYMENT_NAME      | diabetes-aci             |
+| DATASET_NAME      | diabetes_training             |
+| MODEL_NAME      | diabetes_model.pkl             |
```
Comment (dtzar) [inline docs/getting_started.md]: `MODEL_NAME` is already provided in the variables.yml and we're trying to minimize manual entry of variables to the group.  Please remove.

File: docs/getting_started.md [hunk 3]
```diff
@@ -60,6 +60,8 @@ The variable group should contain the following required variables:
 | WORKSPACE_NAME           | mlops-AML-WS             |
 | WORKSPACE_SVC_CONNECTION | aml-workspace-connection |
 | ACI_DEPLOYMENT_NAME      | diabetes-aci             |
+| DATASET_NAME      | diabetes_training             |
```
Comment (dtzar) [inline docs/getting_started.md]: this should be optional, please remove.

File: docs/getting_started.md [hunk 4]
```diff
@@ -125,8 +133,8 @@ Check out the newly created resources in the [Azure Portal](https://portal.azure
...
+**Note:** The training ML pipeline uses a [sample diabetes dataset](https://scikit-learn.org/stable/modules/generated/sklearn.datasets.load_diabetes.html) as training data. To use this, you need to create a **Dataset** in your workspace by uoloading the diabetes.csv file, naming the dataset to match the DATASET_NAME variable you created earlier in ***devopsforai-aml-vg*** variable group. You can upload the data file to the default datastore for simplicity, but best practice is to [create and register a datastore](https://docs.microsoft.com/en-us/azure/machine-learning/how-to-access-data#azure-machine-learning-studio) in your ML workspace and upload the datafile to the corresponding blob container. You can also define a datastore in the ML Workspace with [az cli](https://docs.microsoft.com/en-us/cli/azure/ext/azure-cli-ml/ml/datastore?view=azure-cli-latest#ext-azure-cli-ml-az-ml-datastore-attach-blob).
+You can also configure DATASTORE_NAME and DATAFILE_NAME variables in ***devopsforai-aml-vg*** variable group to have the pipeline automatically upload the data file to your datastore and register a dataset.
```
Comment (dtzar) [inline docs/getting_started.md]: We should be clearer on what to do for dataset for three scenarios with this repo, perhaps a new ## section
1. Leave as-is (None of the DATA* env vars specified) - we use the stock diabetes dataset.  This is only applicable for MLOpsPython diabetes_regression model and will not work for BYOC.
1. Upload your own dataset, specify just `DATASET_NAME`
1. Specify all three DATA* vars

File: docs/getting_started.md (hunk 3 above)
Comment (GraemeMalcolm) [reply docs/getting_started.md]: Why optional? One way or another, they'll need to create a dataset, won't they?

File: docs/getting_started.md (hunk 4 above)
Comment (GraemeMalcolm) [reply docs/getting_started.md]: Rewrote as discussed to generate the diabetes dataset on the fly if no DATASET_NAME is supplied

File: docs/getting_started.md (hunk 2 above)
Comment (GraemeMalcolm) [reply docs/getting_started.md]: See update

File: docs/getting_started.md (hunk 3 above)
Comment (GraemeMalcolm) [reply docs/getting_started.md]: See update

File: diabetes_regression/register/register_model.py (hunk 1 above)
Comment (GraemeMalcolm) [reply diabetes_regression/register/register_model.py]: See updated logic

**Comment by GraemeMalcolm** [comment]
I'm getting a "Bad Gateway" error in the smoke test after the model service is deployed. I don't think this is due to my changes - it looks like an AML service error, but we should probably hold off on merging until we troublehoot it.

File: docs/getting_started.md (hunk 2 above)
Comment (dtzar) [reply docs/getting_started.md]: Code looks good, but we should only be putting values here which are not required to be manually entered.

Comment (GraemeMalcolm) [reply docs/getting_started.md]: I updated the getting started markdown to remove MODEL_NAME and DATASET_NAME. Is there something else that needs to be removed?

Comment (GraemeMalcolm) [reply docs/getting_started.md]: I removed the unused DATA vars from the ENV example

Comment (GraemeMalcolm) [reply docs/getting_started.md]: and from env-variables.py

**Pull request merged**